Meadowlark-js/packages/meadowlark-core/test/handler/Update.test.ts

```diff
@@ -108,6 +108,42 @@ describe('given the new document has an invalid reference ', () => {
   });
 });
 
+describe('given update has write conflict failure', () => {
+  let mockDocumentStore: any;
+  const expectedError = 'Write conflict error returned';
+  let response: FrontendResponse;
+
+  beforeAll(async () => {
+    mockDocumentStore = jest.spyOn(PluginLoader, 'getDocumentStore').mockReturnValue({
+      ...NoDocumentStorePlugin,
+      updateDocumentById: async () =>
+        Promise.resolve({
+          response: 'UPDATE_FAILURE_WRITE_CONFLICT',
+          failureMessage: expectedError,
+        }),
+    });
+
+    // Act
+    response = await update(frontendRequest);
+  });
+
+  afterAll(() => {
+    mockDocumentStore.mockRestore();
+  });
+
+  it('returns status 409', () => {
+    expect(response.statusCode).toEqual(409);
+  });
+
+  it('has a failure message', () => {
+    expect(response.body).toMatchInlineSnapshot(`
+      {
+        "error": "Write conflict error returned",
+      }
+    `);
+  });
+});
+
```

Comment:
As mentioned in the ticket, do we need a test for the case where 404 should be returned if a PUT fails due to a DELETE? Or is it already covered?

Reply:
I don't think there is a way to know that a transaction is failing due to a delete. I mean, Mongo will throw the error when there is a conflict, but it won't tell me if the error occurred because the transaction being processed is a delete or an update. I think I can inspect the mongo error object to double check.